### PR TITLE
Convert dequantize_linear to an internal function _dequantize_linear

### PR DESCRIPTION
**Pull request by nishantpdce**
Stack:
&nbsp;&nbsp;&nbsp;&nbsp;:black_circle:&nbsp; **#20938 Convert dequantize_linear to an internal function _dequantize_linear**&nbsp;&nbsp;[:yellow_heart:](https://our.internmc.facebook.com/intern/diff/D15446097/)

Dequantize_linear need not be exposed to the front end users.
It will only be used for the jit passes for q-dq insertion and op
substitution.

Differential Revision: [D15446097](https://our.internmc.facebook.com/intern/diff/D15446097/)

### CODE REVIEW

**Review by jerryzh168 — APPROVED** [review]

File: docs/source/tensors.rst [hunk 1]
```diff
@@ -209,7 +209,7 @@ view of a storage and defines numeric operations on it.
    .. automethod:: cumsum
    .. automethod:: data_ptr
    .. automethod:: dequantize
-   .. automethod:: dequantize_linear
+   .. automethod:: _dequantize_linear
```
Comment (jerryzh168) [inline docs/source/tensors.rst]: remove this?

File: test/test_jit.py [hunk 2]
```diff
@@ -1469,7 +1469,7 @@ def __init__(self):
             def forward(self, x):
                 x = x.quantize_linear(1.0, 0, torch.uint8)
                 x = x.int_repr()
-                x = x.dequantize_linear(1.0, 0, torch.uint8)
+                x = x._dequantize_linear(1.0, 0, torch.uint8)
```
Comment (jerryzh168) [inline test/test_jit.py]: why do we still have this? should be calling torch._dequantize_linear(...)

File: torch/_tensor_docs.py [hunk 3]
```diff
@@ -3031,9 +3031,9 @@ def callable(a, b) -> number
 See :func:`torch.det`
 """)
 
-add_docstr_all('dequantize_linear',
+add_docstr_all('_dequantize_linear',
```
Comment (jerryzh168) [inline torch/_tensor_docs.py]: remove this as well

**Review by jerryzh168 — CHANGES_REQUESTED** [review]
actually let's remove `_dequantize_linear` as a Tensor method as well.

File: test/test_jit.py (hunk 2 above)
Comment (jerryzh168) [reply test/test_jit.py]: one more here

Comment (nishantpdce) [reply test/test_jit.py]: yeah once this is changed, we can remove it from tensor methods also right

**Review by jerryzh168 — APPROVED** [review]
please make sure to remove the tensor method

**Pull request closed without merging**